From 7215ccb2a70b22035d3803ab97e3525e3116fbb3 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Wed, 8 May 2024 07:04:54 +0200
Subject: [PATCH] Revert "gold: Use char16_t, char32_t instead of uint16_t,
 uint32_t as character types"

This reverts commit 5e9091dab8858b25210a91d22fbbbfdee9c969ad.
---
 gold/merge.cc      | 4 ++--
 gold/output.cc     | 5 ++---
 gold/stringpool.cc | 5 ++---
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/gold/merge.cc b/gold/merge.cc
index 086c7915d..9490c47a1 100644
--- a/gold/merge.cc
+++ b/gold/merge.cc
@@ -665,10 +665,10 @@ template
 class Output_merge_string<char>;
 
 template
-class Output_merge_string<char16_t>;
+class Output_merge_string<uint16_t>;
 
 template
-class Output_merge_string<char32_t>;
+class Output_merge_string<uint32_t>;
 
 #if defined(HAVE_TARGET_32_LITTLE) || defined(HAVE_TARGET_32_BIG)
 template
diff --git a/gold/output.cc b/gold/output.cc
index adbe624e5..5821cac6a 100644
--- a/gold/output.cc
+++ b/gold/output.cc
@@ -29,7 +29,6 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <algorithm>
-#include <uchar.h>
 
 #ifdef HAVE_SYS_MMAN_H
 #include <sys/mman.h>
@@ -2707,10 +2706,10 @@ Output_section::add_merge_input_section(Relobj* object, unsigned int shndx,
 	      pomb = new Output_merge_string<char>(addralign);
 	      break;
 	    case 2:
-	      pomb = new Output_merge_string<char16_t>(addralign);
+	      pomb = new Output_merge_string<uint16_t>(addralign);
 	      break;
 	    case 4:
-	      pomb = new Output_merge_string<char32_t>(addralign);
+	      pomb = new Output_merge_string<uint32_t>(addralign);
 	      break;
 	    default:
 	      return false;
diff --git a/gold/stringpool.cc b/gold/stringpool.cc
index c2b67c472..af62a78bd 100644
--- a/gold/stringpool.cc
+++ b/gold/stringpool.cc
@@ -25,7 +25,6 @@
 #include <cstring>
 #include <algorithm>
 #include <vector>
-#include <uchar.h>
 
 #include "output.h"
 #include "parameters.h"
@@ -528,9 +527,9 @@ template
 class Stringpool_template<char>;
 
 template
-class Stringpool_template<char16_t>;
+class Stringpool_template<uint16_t>;
 
 template
-class Stringpool_template<char32_t>;
+class Stringpool_template<uint32_t>;
 
 } // End namespace gold.
-- 
2.52.0

